src/LaunchDarklyProvider.ts

```diff
@@ -31,20 +37,67 @@ function wrongTypeResult<T>(value: T): ResolutionDetails<T> {
 export default class LaunchDarklyProvider implements Provider {
   private readonly logger: LDLogger;
 
+  private readonly client: LDClient;
+
+  private readonly clientConstructionError: any;
+
   readonly metadata: ProviderMetadata = {
     name: 'launchdarkly-node-provider',
   };
 
+  private innerStatus: ProviderStatus = ProviderStatus.NOT_READY;
+
+  public readonly events = new OpenFeatureEventEmitter();
+
+  /**
+   * Get the status of the LaunchDarkly provider.
+   */
+  public get status() {
+    return this.innerStatus;
+  }
+
   /**
    * Construct a {@link LaunchDarklyProvider}.
    * @param client The LaunchDarkly client instance to use.
    */
-  constructor(private readonly client: LDClient, options: LaunchDarklyProviderOptions = {}) {
+  constructor(sdkKey: string, options: LDOptions = {}) {
     if (options.logger) {
-      this.logger = options.logger;
+      this.logger = new SafeLogger(options.logger, basicLogger({ level: 'info' }));
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
+    try {
+      this.client = init(sdkKey, {
```

Comment:
There are a few cases where the constructor fails. I am capturing it and letting the OF SDK handle it. That seems most in the spirit of the functionality. Maybe a little less convenient though.